features/step_definitions/authenticated_user_steps.rb

```diff
@@ -28,3 +24,14 @@
 Then("show me the page") do
   save_and_open_page
 end
+
+Then("I should be on the Sign in page") do
```

Comment:
This is all fine, but... You will end up with a lot of steps that are doing exactly the same thing. You can refactor this to make it more dynamic. You could have a step like `I should be on the {string} page` and it's implementation would look something like

```ruby
Then("I should be on the {string} page") do |page_name|
  if page_name == 'Sign in'
    path = new_user_session_path
  end

  expect(page.current_path).to eq path
end
```

To make this even better you could extract the page path selection into a method

```ruby
def page_path_from(page_name)
  if page_name == 'Sign in'
    new_user_session_path
  elsif page_name == 'Sign up'
    user_registration_path
  else
    raise "#{page_name} is not defined"
  end
end
```

And you replace your expectation above with

```ruby
Then("I should be on the {string} page") do |page_name|
  expect(page.current_path).to eq page_path_from(page_name)
end
```